apps/ui/src/queries/proposals.ts

```diff
@@ -4,43 +4,49 @@ import {
   useQuery,
   useQueryClient
 } from '@tanstack/vue-query';
-import { MaybeRef } from 'vue';
+import { MaybeRefOrGetter } from 'vue';
 import { getNames } from '@/helpers/stamp';
 import { getNetwork } from '@/networks';
 import { ProposalsFilter } from '@/networks/types';
 import { NetworkID, Proposal } from '@/types';
 
 type Filters = {
-  state?: MaybeRef<NonNullable<ProposalsFilter['state']>>;
-  labels?: MaybeRef<string[]>;
+  state?: MaybeRefOrGetter<NonNullable<ProposalsFilter['state']>>;
+  labels?: MaybeRefOrGetter<string[]>;
 };
 
 export const PROPOSALS_LIMIT = 20;
 export const PROPOSALS_SUMMARY_LIMIT = 6;
 
 export const PROPOSALS_KEYS = {
   all: ['proposals'] as const,
-  space: (networkId: NetworkID, spaceId: string) =>
-    [...PROPOSALS_KEYS.all, `${networkId}:${spaceId}`] as const,
-  spaceSummary: (networkId: NetworkID, spaceId: string) =>
-    [...PROPOSALS_KEYS.space(networkId, spaceId), 'summary'] as const,
+  space: (
+    networkId: MaybeRefOrGetter<NetworkID>,
+    spaceId: MaybeRefOrGetter<string>
+  ) => [...PROPOSALS_KEYS.all, networkId, spaceId] as const,
```

Comment:
Don't need a `toValue` before using those ref ?

Reply:
It seems to work from now, since they're already values since coming from `props.x`, but may cause confusion later when passing a ref.

Shouldn't everything coming from `MaybeRefOrGetter` always converted to value before usage, as best practice ?

Reply:
> It seems to work from now, since they're already values since coming from props.x, but may cause confusion later when passing a ref.

Those should not be values, we are converting our props to refs using `toRef(() => props.space.network)`, otherwise our composables won't be able to react to those props changing.

> Don't need a toValue before using those ref ?
In our current usecase, we have to use refs in keys. If we call toValue it won't be reactive anymore, that means it will be stuck with initial value for the key. `vue-query` will watch those refs to compute actual key reactively.

So if ref changes, key will change and queryFn will fire - in queryFn we can unwrap ref using `toValue`.